tools/donate-cpu.py

```diff
@@ -58,17 +58,27 @@ def get_cppcheck(cppcheck_path):
     print('Get Cppcheck..')
     for i in range(5):
         if os.path.exists(cppcheck_path):
-            os.chdir(cppcheck_path)
-            subprocess.call(['git', 'checkout', '-f'])
-            subprocess.call(['git', 'pull'])
+            try:
+                os.chdir(cppcheck_path)
+                subprocess.check_call(['git', 'checkout', '-f'])
+                subprocess.check_call(['git', 'pull'])
+            except:
+                print('Failed to update Cppcheck sources! Retrying..')
```

Comment:
If `git pull` fails, maybe we can try to remove the cppcheck folder and `git clone` it?

Reply:
or at least you can write a message to the user something like:

    Please remove the whole workfolder 'rm -rf ~/cppcheck-donate-cpu-workfolder' and restart the donate-cpu.py script.

In the extreme case that the file system is corrupt .. then some kind of human fixing is required.

Reply:
> If `git pull` fails, maybe we can try to remove the cppcheck folder and `git clone` it?

Let's try this. Generally I don't like scripts removing things, but I guess in this case it **should** be ok.